src/masternodes/mn_rpc.cpp

```diff
@@ -2877,9 +2877,18 @@ void CheckAndFillPoolSwapMessage(const JSONRPCRequest& request, CPoolSwapMessage
 
             auto resA = pool.reserveA;
             auto resB = pool.reserveB;
-            if (poolSwapMsg.idTokenFrom != pool.idTokenA) {
-                std::swap(resA, resB);
+
+            // NOTE it has a bug prior Dakota hardfork
+            if (height < Params().GetConsensus().DakotaHeight) {
+                if (poolSwapMsg.idTokenFrom != pool.idTokenA) {
+                    std::swap(resA, resB);
+                }
```

Comment:
We don't need hardfork check on RPC side

Reply:
If maxPrice is null just set both to int MAX, then it has no protection @uzyn asked, then the app can pass in max price

Reply:
Done.